src/substrait/textplan/parser/SubstraitPlanRelationVisitor.cpp

```diff
@@ -1243,160 +1482,37 @@ ::substrait::proto::Expression_Literal SubstraitPlanRelationVisitor::visitTime(
-      } catch (...) {
-        errorListener_->addError(token, "Could not parse varchar length.");
-      }
-      type.mutable_varchar()->set_nullability(nullValue);
-      break;
-    }
-    case TypeKind::kFixedBinary:
-      type.mutable_fixed_binary()->set_nullability(nullValue);
-      break;
-    case TypeKind::kDecimal: {
-      auto dec = reinterpret_cast<const ParameterizedDecimal*>(&decodedType);
-      if (dec == nullptr) {
-        break;
-      }
-      try {
-        int32_t precision = std::stoi(dec->precision()->value());
-        int32_t scale = std::stoi(dec->scale()->value());
-        type.mutable_decimal()->set_precision(precision);
-        type.mutable_decimal()->set_scale(scale);
-      } catch (...) {
-        errorListener_->addError(
-            token, "Could not parse decimal precision and scale.");
-      }
-      type.mutable_decimal()->set_nullability(nullValue);
-      break;
-    }
-    case TypeKind::kStruct: {
-      auto structure =
-          reinterpret_cast<const ParameterizedStruct*>(&decodedType);
-      for (const auto& t : structure->children()) {
-        *type.mutable_struct_()->add_types() = typeToProto(token, *t);
-      }
-      type.mutable_struct_()->set_nullability(nullValue);
-      break;
-    }
-    case TypeKind::kList: {
-      auto list = reinterpret_cast<const ParameterizedList*>(&decodedType);
-      *type.mutable_list()->mutable_type() =
-          typeToProto(token, *list->elementType());
-      type.mutable_list()->set_nullability(nullValue);
-      break;
-    }
-    case TypeKind::kMap: {
-      auto map = reinterpret_cast<const ParameterizedMap*>(&decodedType);
-      if (map->keyType() == nullptr || map->valueType() == nullptr) {
-        errorListener_->addError(
-            token, "Maps require both a key and a value type.");
-        break;
-      }
-      *type.mutable_map()->mutable_key() = typeToProto(token, *map->keyType());
-      *type.mutable_map()->mutable_value() =
-          typeToProto(token, *map->valueType());
-      type.mutable_map()->set_nullability(nullValue);
-      break;
-    }
-    case TypeKind::kKindNotSet:
-      errorListener_->addError(token, "Unable to recognize requested type.");
-      break;
+int32_t SubstraitPlanRelationVisitor::visitSortDirection(
+    SubstraitPlanParser::IdContext* ctx) {
+  std::string text = normalizeProtoEnum(ctx->getText(), kSortDirectionPrefix);
+  if (text == "unspecified") {
+    return ::substrait::proto::SortField::SORT_DIRECTION_UNSPECIFIED;
+  } else if (text == "ascnullsfirst") {
+    return ::substrait::proto::SortField::SORT_DIRECTION_ASC_NULLS_FIRST;
+  } else if (text == "ascnullslast") {
+    return ::substrait::proto::SortField::SORT_DIRECTION_ASC_NULLS_LAST;
+  } else if (text == "descnullsfirst") {
+    return ::substrait::proto::SortField::SORT_DIRECTION_DESC_NULLS_FIRST;
+  } else if (text == "descnullslast") {
+    return ::substrait::proto::SortField::SORT_DIRECTION_DESC_NULLS_LAST;
```

Comment:
I don't think we need to do this but maybe we should break apart the asc/desc and the nullsfirst/nullslast?  For example, could a user type "asc" (and it would default to nulls first?) or "nullslast" (and it would default to whatever direction)

Reply:
Added issue #69 for this.